### PR TITLE
[closes #39] Auto select event week.

### LINKED ISSUE #39: Default Events view to current week
We might be able to make use of this helper method to clean up the groupEventByWeeks method https://github.com/the-blue-alliance/the-blue-alliance-android/blob/c4948c3a560e3b73833197b92d24295027004afe/android/src/main/java/com/thebluealliance/androidclient/helpers/EventHelper.java#L98

### PR DESCRIPTION
**Pull request by brandondean**
Created an `EventWeek` class that holds a number of static methods (err class methods) that
determines what event week is for a given date. `EventViewController` uses this new class to
pre-select the current event week when first visiting event list.

Admittedly these methods are basically a complete rip off of what the android app has.
https://github.com/the-blue-alliance/the-blue-alliance-android/blob/c4948c3a560e3b73833197b92d24295027004afe/android/src/main/java/com/thebluealliance/androidclient/helpers/EventHelper.java#L98

https://github.com/the-blue-alliance/the-blue-alliance-android/blob/c4948c3a560e3b73833197b92d24295027004afe/android/src/main/java/com/thebluealliance/androidclient/helpers/EventHelper.java#L98

### CODE REVIEW

File: the-blue-alliance-ios/EventsViewController.m [hunk 1]
```diff
@@ -37,10 +38,8 @@ - (void)setWeeks:(NSArray<NSNumber *> *)weeks {
     _weeks = weeks;
     
     if (weeks && !self.currentWeek) {
-        NSNumber *week = weeks.firstObject;
-        
-        self.currentWeek = week;
-        self.eventsViewController.week = week;
+        self.currentWeek = @([EventWeek eventOrderForDate:[NSDate date]]);
```
Comment (ZachOrr) [inline the-blue-alliance-ios/EventsViewController.m]: If this year is not the current year, can we default to the first week of competition?

**Comment by ZachOrr** [comment]
I know Android also fixed a bug with "gaps" (for the week between Week 7 and CMP) - let's make sure we take that in to account https://github.com/the-blue-alliance/the-blue-alliance-android/issues/699

**Comment by brandondean** [comment]
@ZachOrr Ah. That's what https://github.com/the-blue-alliance/the-blue-alliance-ios/pull/48/files#diff-d5b668550a4184b0b16d17f7aeccbd83R285 is for.

File: the-blue-alliance-ios/EventsViewController.m [hunk 2]
```diff
@@ -36,9 +36,11 @@ @implementation EventsViewController
 
 - (void)setWeeks:(NSArray<NSNumber *> *)weeks {
     _weeks = weeks;
-    
+
     if (weeks && !self.currentWeek) {
-        self.currentWeek = @([EventWeek eventOrderForDate:[NSDate date]]);
+        self.currentWeek = self.currentYear.integerValue == [[NSCalendar currentCalendar] component:NSCalendarUnitYear fromDate:[NSDate date]] ?
```
Comment (ZachOrr) [inline the-blue-alliance-ios/EventsViewController.m]: This won't work for all years pre-1995, where CMP is the only event

Comment (brandondean) [inline the-blue-alliance-ios/EventsViewController.m]: Ah yes. Good catch.

File: the-blue-alliance-ios/Models/Event.m [hunk 3]
```diff
@@ -328,18 +343,24 @@ + (NSInteger)competitionWeekForDate:(NSDate *)date {
 }
 
 
-+ (NSArray *)championshipWeeks {
++ (NSArray *)championshipCompetitionWeeks {
```
Comment (ZachOrr) [inline the-blue-alliance-ios/Models/Event.m]: Competiion 😅

**Comment by ZachOrr** [comment]
Looks good 👍

**Pull request merged**